src/logger.ts

```diff
@@ -26,36 +37,37 @@ function Logger(config) {
         }
     };
 
-    this.error = function(msg) {
+    this.error = function(msg: string) {
         if (logLevel !== 'none') {
             if (self.logger.error) {
                 self.logger.error(msg);
             }
         }
     };
 
-    this.setLogLevel = function(newLogLevel) {
+    this.setLogLevel = function(newLogLevel: string) {
         logLevel = newLogLevel;
+        this.logLevel = logLevel;
```

Comment:
I think you should be able to get rid of the redundant assignment and simplify this:

```
this.setLogLevel = function(_logLevel: string) {
        this.logLevel = _logLevel;
```